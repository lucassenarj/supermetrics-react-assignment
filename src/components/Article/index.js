import React from "react";
import PropTypes from "prop-types";
import { formatDate } from "./../../utils/helpers";
import "./index.scss";

function Article({date, message}) {
  return (
    <article>
      <div className="article__header">
        <p>{ formatDate(date) }</p>
      </div>
      <div className="article__content">
        <p>{ message }</p>
      </div>
    </article>
  )
};

Article.propTypes = {
  date: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
};

export default Article;
