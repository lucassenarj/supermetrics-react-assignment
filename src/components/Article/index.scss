article {
  width: 100%;
  display: flex;
  flex-direction: column;
  margin: 20px auto;
  border: solid 2px #cc9cbd;
}

.article__header {
  padding: 10px;
  background-color: #efd7e8;
  border-bottom: solid 2px #cc9cbd;
}

.article__content {
  padding: 10px;
  background-color: #e2bbd6;
}