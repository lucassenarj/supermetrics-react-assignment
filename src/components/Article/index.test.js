import React from "react";
import { render } from "@testing-library/react";
import Article from "./";

const values = {
  date: "2021-06-08T16:27:02+00:00",
  message: "Some cool messages",
}

describe("Components", () => {
  describe("Article", () => {
    it("renders correctly", () => {
      const { container } = render(<Article date={values.date} message={values.message} />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});