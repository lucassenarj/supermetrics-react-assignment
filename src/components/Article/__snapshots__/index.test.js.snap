// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Article renders correctly 1`] = `
<article>
  <div
    class="article__header"
  >
    <p>
      June 8, 2021, 17:27:02
    </p>
  </div>
  <div
    class="article__content"
  >
    <p>
      Some cool messages
    </p>
  </div>
</article>
`;
